translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -6510,6 +6538,31 @@ Function *SPIRVToLLVM::transFunction(SPIRVFunction *BF) {
       transValue(BInst, F, BB, false);
     }
   }
+
+  // Update phi nodes -- add missing incoming arcs.
+  // This is necessary because LLVM's CFG is a multigraph, while SPIR-V's
+  // CFG is not.
+  for (BasicBlock &BB : *F) {
+    DenseMap<BasicBlock*, unsigned> PredecessorsToAdd =
+      getBlockPredecessorCounts(&BB);
+
+    if (PredecessorsToAdd.empty())
+        continue;
+
+    // Add missing incoming arcs to each phi node that requires fixups.
+    for (PHINode &Phi : BB.phis()) {
+      const unsigned InitialNumIncoming = Phi.getNumIncomingValues();
+      for (unsigned i = 0; i != InitialNumIncoming; ++i) {
+        BasicBlock* Predecessor = Phi.getIncomingBlock(i);
```

Comment:
what will happen if there are duplicated Predecessor in original Phi incoming blocks?

Reply:
SPIR-V doesn't allow it, so it should not ever happen.

Reply:
I'll add an assertion for this case.